src/common/api-requests/user.ts

```diff
@@ -0,0 +1,13 @@
+import { api } from 'boot/axios';
+
+import { ParseableCourseUser } from 'src/common/models/users';
+import { ResponseError } from 'src/common/api-requests/interfaces';
+
+export async function checkIfUserExists(course_id: number, username: string) {
+	const url = `courses/${course_id}/users/${username}/exists`;
```

Comment:
A general rule of thumb in programming for any language is that if a variable is used only once, it probably should not exist.  Instead just use its value in the one place the variable is used.

You do this a lot.  There are many reasons for this rule.  First is efficiency.  Adding a variable makes the program have to allocate space in the stack for it and then perform a lookup when it is used.  The next reason is readability.  The logic here is actually the same as that for the efficiency reason.  This makes someone reading the code have to look back to find what the variable is.  Yet another reason (seen quite frequently in the code for webwork 2 where this bad programming practice is prevalent) is that as the code evolves that one usage may be removed.  If the coder that removes that usage is not paying attention, the variable definition gets left behind.

I see several places where it seems that you might have done this to prevent long lines.  For example, on lines 30-31 and 36-37 of src/common/api-requests/library.ts.  That is not a valid reason to create a new variable.  Find a way to wrap the line appropriately.  In both of those cases I am easily able to do this.

Reply:
> I see several places where it seems that you might have done this to prevent long lines. For example, on lines 30-31 and 36-37 of src/common/api-requests/library.ts. That is not a valid reason to create a new variable. Find a way to wrap the line appropriately. In both of those cases I am easily able to do this.

This is exactly the reason I usually have done this.  I can remove the variable and make the line length okay by wrapping.

Reply:
Sounds good.  Note that one convenience of es6 interpolated strings is that you can always break a line inside of a `${...}`.  For example for lins 36-38 of library.ts you can change those to 
```typescript
	return (await axios.get(`/opl/api/taxo/disciplines/${lib_id.disc_id}/subjects/${
		lib_id.subj_id ?? 0}/chapters`)).data as Array<LibraryCategory>;
```
Note I took this even further and removed the `const response` variable as well.  You don't need to go that far.  Just nothing that this can be taken further.